### ISSUE TITLE
Regression: Can only scroll about one screen up before it stops

### ISSUE THREAD

**Issue by aral** [helix-editor/helix#6397]
### Summary

When scrolling with the mouse wheel, I can only scroll up for about a screenful before it stops. I then have to click in the page again before I can scroll another screenful using the mouse wheel.

I noticed this after updating to the latest Helix today but I hadn’t updated in a bit so I don’t know exactly when the regression occurred.

### Reproduction Steps

1. Open a long piece of code

2. Scroll down using the mouse wheel and notice that you can scroll without problems as much as you like (on my Logitech mouse, the mouse wheel as a mode where it spins freely so I can flick it, watch it keep scrolling and stop it when I’m where I want to be to quickly scroll).

3. Scroll up using the mouse wheel and notice that it stops after about a screen’s worth of scrolling.

## Workaround

Click anywhere in the terminal and scroll again. Rinse and repeat for every page you want to scroll up.

### Helix log

No log statements output while scrolling.

### Platform

Linux

### Terminal Emulator

Tested in latest WezTerm and latest GNOME Console (it’s not the terminal)

### Helix Version

22.12-443-g78613ac0

**Comment by pascalkuthe** [comment]
I can not reproduce this. What settings are you using?

**Comment by aral** [comment]
@pascalkuthe I’m running on latest Fedora Silverblue (GNOME under Wayland)

## Config file:

```toml
theme = "onelight"

[editor]
soft-wrap.enable = true
scroll-lines = 1
idle-timeout = 0
bufferline = "multiple"
color-modes = true
cursorline = true
auto-format = false
# rulers = [80]
# gutter = ["diagnostics", "spacer", "line-numbers", "spacer", "diff"]

[editor.statusline]
right = ["spinner", "selections", "mode"]
left = ["diagnostics", "file-name", "file-encoding", "position", "position-percentage"]
mode.normal = "NORMAL"
mode.insert = "INSERT"
mode.select = "SELECT"

[editor.indent-guides]
render = true

[editor.cursor-shape]
insert = "bar"

[editor.lsp]
auto-signature-help = false

[editor.whitespace.render]
space = "all"
tab = "all"
newline = "none"

[keys.normal]
C-s = ":write"
C-q = ":quit"
"'" = "save_selection"
S-tab = "jump_backward"
C-w = ":buffer-close"
C-z = "undo"
"C-a" = "signature_help"
C-x = "increment"
C-S-x = "decrement"

[keys.insert]
up = ["normal_mode", "move_line_up"]
down = ["normal_mode", "move_line_down"]
left = "move_char_left"
right = "move_char_right"
C-left = "move_prev_word_end"
C-right = "move_next_word_start"
end = "goto_line_end_newline"
home = "goto_line_start"
j = {k = "normal_mode"}
"C-a" = "signature_help"

[keys.select]
C-s = ":write"

[editor.file-picker]
hidden = false
parents = true
```

**Comment by aral** [comment]
Here’s a video demonstrating the problem.

Between each click, I’m free spinning the scroll wheel to try and get to the top of the page. Previously, it would just freely scroll to the top.

[Screencast from 2023-03-24 09-52-20.webm](https://user-images.githubusercontent.com/41798/227488515-673cb4e7-0772-4975-a343-13ffc596641f.webm)

To contrast, here’s what scrolling down by free-spinning the mouse wheel (which works the same as it did previously) looks like:

[Screencast from 2023-03-24 09-54-37.webm](https://user-images.githubusercontent.com/41798/227489149-8a88a4d5-642f-4fd1-b276-fffce6663414.webm)

**Comment by pascalkuthe** [comment]
Ah interesting I did discover further bug here. Please try with #6417. If that fixes thebjssue I will try to back port that specific fix

**Comment by aral** [comment]
@pascalkuthe I can’t reproduce the bug with your inline-diagnostics branch (https://github.com/pascalkuthe/helix/tree/inline-diagnostics)

**Comment by pascalkuthe** [comment]
Great so I did stumble across the same bug :D I am going to post a PR with smaller fixes extracted from that branch to before the next release. I will include the fix for the scrolling. O will let you know once that PR is ready. Would be great if you could test again since I can really reproduce locally

**Comment by aral** [comment]
> Would be great if you could test again since I can't really reproduce locally

Happy to :)

**Comment by pascalkuthe** [comment]
Should be fixed by #6440, could you test to confirm?

**Comment by aral** [comment]
@pascalkuthe Can confirm that I cannot reproduce the scroll wheel issue with e6b386f2 from the cherry-pick branch in #6640. Thanks, Pascal :)
